### DISCUSSION TITLE
Changing Selectors while another key is held will press that key again. Can I turn it off?

### DISCUSSION THREAD

**Discussion by Hardivh** [AntiMicroX/antimicrox#207] [Q&A]
**Question**

Sorry I don't know if I'll be able to explain my issue, and english is not my first language, so please bear with me :(

I'm using antimicro and made a profile with 3 sets. One "normal", and 2 other Sets with the "while held" option.

I'm using a PSX gamepad so I'm basically using L1 and R1 as modifiers (set changers while held)

The problem is that when I change subsets (either holding or releasing L1 for instance) with another button held (like square for instace) the PC understands that a new input was made. I want to turn this off because it's affecting my gameplay.

So, to simplify.
My gamepad's "Square" button is "X" on my keyboard on Set1 and "Q" on Set2
My gamepad's "L1" button goes to Set2 while held

When I hold "L1" and press (and hold) "Square", I get the correct input "Q".
But If I release "L1" and keep "Square" held, it autimatically inputs "X" and I don't want that.

I only want inputs to be made when "Square" is pressed, not when held/released, is there a way to do so? I tought the "toggle" option would do that but maybe it doesn't work when Sets are changed.

So if I go on my notepad and hold "Square" while holding and pressing "L1" I get the following:
XQXQXQXQXQXQXQXQ....
I don't want that. It should just input a single "X" and then nothing else until I actually press the "Square" button again...

I know a simpler solution would be changing "L1" to "shift" and then changing my ingame keybing of "Q" to "Shift+X", but the game unfortunatelly doesn't support that (DFO) so yeah, it's screwing me hehehe

Can you guys help me? I hope it's not too confusing.
Thanks in advance!

**Comment by dlip** [comment]
I'm having the same issue as you. I don't have an answer for you unfortunately. I've tried changing the button to trigger on release, but unfortunately switching between sets also triggers a release. I was trying to use layers on joysticks also using the 'while held' option, and its easy to get stuck in layers if you try to use diagonals.

I think the issue is that sets are a poor substitute for a proper combo feature.

**Comment by Hardivh** [comment]
So, guess there's nothing to do about it?

**Comment by mrbeauregarde** [comment]
I don't understand why you cant just let go of square first?

Use Cycle, depending on how long the interact is, set cycle reset
Use Pause 0.00s

Use 2 way shifts and map the rest of set 1 to 2 where possible

Theres plenty to do about it, but look around, this is a place of hobbyists maintaining a thing you use for free. If all you got is the one thing i tried and hands out waiting, I'd expect to keep waiting. Try a bunch of techniques, read up, seek knowledge rather than answers, and youll be liable to get the answer yourself
